hgtv/views/video.py

```diff
@@ -534,26 +438,26 @@ def video_delete(channel, playlist, video):
     (Playlist, {'name': 'playlist', 'channel': 'channel'}, 'playlist'),
     (Video, {'url_name': 'video'}, 'video'),
     permission='remove-video')
+@render_with({'text/html': 'index.html.jinja2'}, json=True)
 def video_remove(channel, playlist, video):
     """
     Remove video from playlist
     """
+    if request.method == 'GET':
+        return {'playlist': dict(playlist.current_access()), 'video': dict(video.current_access())}
     if playlist not in video.playlists:
-        # This video isn't in this playlist
-        abort(404)
+        return make_response(jsonify(status='error', errors={'error': ['Video not playlist and cannot be removed']}), 400)
 
     # If this is the primary playlist for this video, refuse to remove it.
     if playlist == video.playlist:
-        return render_message(title="Cannot remove",
-            message=Markup("Videos cannot be removed from their primary playlist. "
-                '<a href="%s">Return to video</a>.' % video.url_for()))
-
-    connection = PlaylistVideo.query.filter_by(playlist_id=playlist.id, video_id=video.id).first_or_404()
-
-    return render_delete_sqla(connection, db, title="Confirm remove",
-        message=u"Remove video '%s' from playlist '%s'?" % (video.title, playlist.title),
-        success=u"You have removed video '%s' from playlist '%s'." % (video.title, playlist.title),
-        next=playlist.url_for())
+        return make_response(jsonify(status='error', errors={'error': ['Videos cannot be removed from their primary playlist']}), 400)
+    form = Form()
+    if form.validate_on_submit():
+        connection = PlaylistVideo.query.filter_by(playlist_id=playlist.id, video_id=video.id).first_or_404()
+        db.session.delete(connection)
```

Comment:
Please confirm that `PlaylistVideo` has no delete cascade to `Playlist` and `Video`.

Reply:
Doesn't look like that. `PlaylistVideo` has `delete-orphan` cascade configured with `Playlist` and `Video`. So only a playlistvideo has the chance of getting deleted.